### PR TITLE
haskellPackages.sensei: patch flaky test

### PR DESCRIPTION
**Pull request by hypevhs**
## Description of changes

Fixed sensei broken build by adding a patch for the test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Set up to merge into #266650

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1046,13 +1046,16 @@ self: super: {
   # test suite requires git and does a bunch of git operations
   restless-git = dontCheck super.restless-git;
 
-  # requires git at test-time *and* runtime, but we'll just rely on users to
-  # bring their own git at runtime. Additionally, sensei passes `-package
-  # hspec-meta` to GHC in the tests, but doesn't depend on it itself.
-  sensei = overrideCabal (drv: {
+  # patch out a flaky test that depends on output from hspec >= v2.11.7.
+  # https://github.com/hspec/sensei/issues/125
+  sensei = appendPatch ./patches/sensei.patch (overrideCabal (drv: {
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1051]: We generally try to pull patches from upstream if at all possible.  There should be a few examples of `fetchpatch` in this file showing creative ways of pulling from individual commits or PRs (even targeting individual files within a commit).

Would it be possible to change this to a use of `fetchpatch`?

Comment (libjared) [reply pkgs/development/haskell-modules/configuration-common.nix:1051]: This is a new issue, so a commit that fixes the failing build does not exist in upstream yet. I opened the issue there.
hspec/sensei#125

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:1051]: Maybe you could send a PR to `sensei` showing sol what you're expecting the change to be?  And then you can pull in the diff from that PR with `fetchpatch`?

Alternatively, maybe you could just set `sensei` as `dontCheck`?  You could argue that the bounds on `hspec` are incorrect in `sensei.cabal` (which we often work around here in Nixpkgs just by setting `dontCheck` when it affects tests).

**Comment by libjared** [comment]
How about this?

**Comment by cdepillabout** [comment]
I was getting hash mismatches when trying to build locally to test:

```
error: hash mismatch in fixed-output derivation '/nix/store/s05rzrdad9rd2wig3cyvzj0kny33r7hh-5c11026fa48e13ea1c351ab882765eb0966f2e97.patch.drv':
         specified: sha256-zmHZ3pjNxfzTCkP728GYjz0ok2rtjCcQNh5rBZ+6AVg=
            got:    sha256-eUCDvypj2bxTRnHLzrcembLMKHg5c3W3quNfclBDsso=
```

I've fixed this and tried pushing to your branch, but that didn't work for some reason, so I instead pushed directly to `haskell-updates`.  I've replace your replacing commit be36bce31c5b with d1235fcaecc88dfb6, which has been merged into `haskell-updates` in https://github.com/NixOS/nixpkgs/pull/266650 as merge commit 0126d4c46f0806.

Can you try building `sensei` from the new top of `haskell-updates` just to make sure that it looks good?

**Comment by libjared** [comment]
Oh, apparently fetchpatch strips out the entire header. I made a mistake in thinking it was suitable to use alongside nix-prefetch-url. Rookie mistake!

Anyway, haskell-updates' sensei builds and runs great. Thanks.

**Pull request closed without merging**